clients/unshaded/src/main/java/tachyon/client/file/options/CreateFileOptions.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file.options;
+
+import com.google.common.base.Throwables;
+
+import tachyon.Constants;
+import tachyon.annotation.PublicApi;
+import tachyon.client.ClientContext;
+import tachyon.client.TachyonStorageType;
+import tachyon.client.UnderStorageType;
+import tachyon.client.WriteType;
+import tachyon.client.file.policy.FileWriteLocationPolicy;
+import tachyon.thrift.CreateTOptions;
+import tachyon.util.CommonUtils;
+
+@PublicApi
+public final class CreateFileOptions {
+  private boolean mRecursive;
+  private long mBlockSizeBytes;
+  private FileWriteLocationPolicy mLocationPolicy;
+  private TachyonStorageType mTachyonStorageType;
+  private long mTTL;
+  private UnderStorageType mUnderStorageType;
+
+  /**
+   * @return the default {@link OutStreamOptions}
+   */
+  public static CreateFileOptions defaults() {
+    return new CreateFileOptions();
+  }
+
+  /**
+   * Creates a new instance with defaults from the configuration.
+   */
+  private CreateFileOptions() {
+    mRecursive = false;
+    mBlockSizeBytes = ClientContext.getConf().getBytes(Constants.USER_BLOCK_SIZE_BYTES_DEFAULT);
+    try {
+      mLocationPolicy =
```

Comment:
I followed the style set in the original code. My auto formatter changes it slightly